app.js

```diff
@@ -34,6 +38,27 @@ function disablesigs(){
   }
 }
 
+function getProxyAgentFromUrl(request_url, request_options=false) {
+  var result = false;
+  var protocol = new URL(request_url).protocol;
+  var proxy_url = getProxyForUrl(request_url);
+  if(proxy_url) {
+    if(protocol == 'https:') {
+      result = new HttpsProxyAgent(proxy_url)
```

Comment:
This should be assigned to const agent. Then the requests can immediately return instead of assigning to the result variable

Reply:
Could you add a Markdown suggestion to clarify on this a little more? I sadly do not have a fluent Node JS knowledge.

Reply:
So are we talking about something like this:

```js
function getProxyAgentFromUrl(request_url, request_options=false) {
  const proxy_url = getProxyForUrl(request_url);
  if(!proxy_url) {
    return false;
  }

  const protocol = new URL(request_url).protocol;
  let agent;

  if (protocol === 'https:') {
    agent = new HttpsProxyAgent(proxy_url);
    return request_options ? { httpsRequestOptions: { agent } } : agent;
  } else if (protocol === 'http:') {
    agent = new HttpProxyAgent(proxy_url);
    return request_options ? { httpRequestOptions: { agent } } : agent;
  }

  return false;
}
```